File: BACK-END/KuentecoChat/src/main/java/com/example/back_end/dto/response/ai/FinancialHealthScoreDTO.java
package com.example.back_end.dto.response.ai;

import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class FinancialHealthScoreDTO {
    private int score; // 0 - 100
    private String grade; // Ej: "Bueno", "Aceptable", "Crítico"
    private List<String> suggestions;
}

File: FRONT-END/lib/dto/chat/response/financial_health_score_dto.dart
class FinancialHealthScoreDTO {
  final int score; // 0-100
  final String grade;
  final List<String> suggestions;

  FinancialHealthScoreDTO({
    required this.score,
    required this.grade,
    required this.suggestions,
  });

  factory FinancialHealthScoreDTO.fromJson(Map<String, dynamic> json) {
    return FinancialHealthScoreDTO(
      score: json['score'] as int,
      grade: json['grade'] as String,
      suggestions: List<String>.from(json['suggestions'] as List),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'score': score,
      'grade': grade,
      'suggestions': suggestions,
    };
  }
}